cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -46,7 +46,10 @@ endif()
 
 # In order for check_function_exists() detection to work, we must signal it to pass an additional flag, which causes the compilation
 # to abort if linking results in any undefined symbols. The CMake detection mechanism depends on the undefined symbol error to be raised.
-set(CMAKE_REQUIRED_FLAGS "-s ERROR_ON_UNDEFINED_SYMBOLS=1")
+# Disable wasm in cmake checks so that (1) we do not depend on wasm support just for configuration (perhaps the user does not intend
+# to build to wasm; using asm.js only depends on js which we need anyhow), and (2) we don't have issues with a separate .wasm file
+# on the side, async startup, etc..
+set(CMAKE_REQUIRED_FLAGS "-s ERROR_ON_UNDEFINED_SYMBOLS=1 -s WASM=0")
```

Comment:
This looks good for now, although in the future we will definitely want to remove this when wasm-capable Node.jses are widely available - we do already pass `-DCMAKE_CROSSCOMPILING_EMULATOR` to point to Node.js, so as long as user has a wasm-capable Node.js I believe it should just work to set `-s WASM=1` (?)

Reply:
Widespread Node support is already there, except for those on v6 LTS. And those using emsdk will be given a recent enough Node version through that.